inc/class-mltools-gutenberg-config-generator.php

```diff
@@ -0,0 +1,478 @@
+<?php
+
+class MLTools_Gutenberg_Config_Generator {
+
+	const META_BOX_ID             = 'wpml-meta-box-id';
+	const META_BOX_TITLE          = 'WPML XML Configuration';
+	const TEXT_DOMAIN             = 'wpml-compatibility-test-tools';
+	const NON_TRANSLATABLE_BLOCKS = [];
+	const WILDCARD_ATTRIBUTES     = [
+		'desktop',
+	];
+
+	public function __construct() {
+		add_action( 'add_meta_boxes', [ $this, 'addMetaBox' ] );
+	}
+
+	/**
+	 * Adds a meta box to the post and page editor.
+	 */
+	public function addMetaBox() {
+		if ( ! use_block_editor_for_post( get_the_ID() ) ) {
+			return;
+		}
+
+		add_meta_box(
+			self::META_BOX_ID,
+			__( self::META_BOX_TITLE, self::TEXT_DOMAIN ),
+			[ $this, 'renderMetaBoxHtml' ],
+			[ 'post', 'page' ]
+		);
+	}
+
+	/**
+	 * Renders the HTML content for the meta box.
+	 *
+	 * @param WP_Post $post The current post object.
+	 */
+	public function renderMetaBoxHtml( $post ) {
+		$content    = get_post_field( 'post_content', $post->ID );
+		$xmlContent = $this->generateXmlFromContent( $content );
+
+		$this->renderMetaBoxHeader();
+		$this->renderXmlTextarea( $xmlContent );
+		$this->renderDebugInformation( $content );
+	}
+
+	/**
+	 * Renders the header for the meta box.
+	 */
+	private function renderMetaBoxHeader() {
+		echo '<h3>' . __( 'WPML: Gutenberg Blocks', self::TEXT_DOMAIN ) . '</h3>';
```

Comment:
```suggestion
		echo '<h3>' . __( 'WPML: Gutenberg Blocks', 'wpml-compatibility-test-tools' ) . '</h3>';
```

@diiegopereira, we cannot use constants for gettext domain.

I also recommend (as a default guideline) to escape translated strings:

```suggestion
		echo '<h3>' . esc_html__( 'WPML: Gutenberg Blocks', 'wpml-compatibility-test-tools' ) . '</h3>';
```